saltboot-formula/_states/saltboot.py

```diff
@@ -491,7 +491,15 @@ def disk_partitioned(name, data):
     # call _compute_start_end early to check disk size
     part = _compute_start_end(data['partitions'], disk_size)
 
-    ok, part, to_delete = _check_existing(part, existing, device)
+    ok = False
+    to_delete = []
+
+    # if repartition is forced, reset to_delete and remove all existing
+    if __grains__.get('saltboot_force_repartition', False):
+        for d in sorted(existing['partitions']):
+            to_delete.append(d)
+    else:
+        ok, part, to_delete = _check_existing(part, existing, device)
 
```

Comment:
I think here is too late. force_repartition should create new partition table - the check should be already at line 454 and it can be simpler than this.

Reply:
You are right. I'll update PR